avm/src/vm/memory.rs

```diff
@@ -233,28 +309,33 @@ impl HandlerMemory {
     }
   }
 
-  /// Pushes raw `mems` indexes into a fractal at a given address. Allows pointers between fractal
-  /// data when no explicit address exists between them.
-  pub fn push_idxs(self: &mut HandlerMemory, addr: i64, a: usize, b: usize) {
+  /// Pushes a pointer to an address from a FractalMemory
+  pub fn push_register_from_fractal(self: &mut HandlerMemory, addr: i64, fractal: &FractalMemory, idx: usize) {
     let mem = self.read_mut_fractal(addr);
-    mem.push((a, b as i64));
+    mem.push(fractal.block[idx]);
   }
 
   /// Pops a value off of the fractal. May be fixed data or a virtual pointer.
-  pub fn pop(self: &mut HandlerMemory, addr: i64) -> Result<(usize, i64), String> {
+  pub fn pop(self: &mut HandlerMemory, addr: i64) -> Result<FractalMemory, String> {
     let mem = self.read_mut_fractal(addr);
     if mem.len() > 0 {
-      return Ok(mem.pop().unwrap());
+      return Ok(FractalMemory{
+        block: vec![mem.pop().unwrap()],
+        hm_addr: Some(addr),
```

Comment:
Since it was popped off of the array, should it actually be getting the address here?

Reply:
good point!